### ISSUE TITLE
Bookmarks not being overwritten, gets 'undefined' nickname

### ISSUE THREAD

**Issue by guusdk** [conversejs/converse.js#2190]
Using Converse 6.0.1, I got this behavior. On a brand new Openfire instance, I've created a MUC room, logged in with Conversations (the Android client) and set a bookmark.

Server-sided, this stanza was received:

```
<iq from="admin@laptop-guus/Conversations.drRQ" id="g7XvAucXKBnC" type="set">
    <pubsub xmlns="http://jabber.org/protocol/pubsub">
        <publish node="storage:bookmarks">
            <item id="current">
                <storage xmlns="storage:bookmarks">
                    <conference autojoin="true" jid="testroom@conference.laptop-guus"/>
                </storage>
            </item>
        </publish>
        <publish-options>
            <x type="submit" xmlns="jabber:x:data">
                <field type="hidden" var="FORM_TYPE">
                    <value>http://jabber.org/protocol/pubsub#publish-options</value>
                </field>
                <field var="pubsub#access_model">
                    <value>whitelist</value>
                </field>
                <field var="pubsub#persist_items">
                    <value>true</value>
                </field>
            </x>
        </publish-options>
    </pubsub>
</iq>
```

Next, I logged in with the same account into Conversations (6.0.1). I've joined the room through the bookmark that was visible in the UI, which caused the UI to prompt me for a nickname. When I provided that, I clicked the 'bookmark' icon, which allowed me to set the bookmark again. When doing that, server-sided, I received this:

```
<iq from="admin@laptop-guus/converse.js-109250898" id="f869fc80-f202-4e78-874f-cfe49b666fb7:sendIQ" type="set">
    <pubsub xmlns="http://jabber.org/protocol/pubsub">
        <publish node="storage:bookmarks">
            <item id="current">
                <storage xmlns="storage:bookmarks">
                    <conference autojoin="true" jid="testroom@conference.laptop-guus" name="testroom">
                        <nick>Guus</nick>
                    </conference>
                    <conference autojoin="true" jid="testroom@conference.laptop-guus" name="testroom@conference.laptop-guus">
                        <nick>undefined</nick>
                    </conference>
                </storage>
            </item>
        </publish>
        <publish-options>
            <x type="submit" xmlns="jabber:x:data">
                <field type="hidden" var="FORM_TYPE">
                    <value>http://jabber.org/protocol/pubsub#publish-options</value>
                </field>
                <field var="pubsub#persist_items">
                    <value>true</value>
                </field>
                <field var="pubsub#access_model">
                    <value>whitelist</value>
                </field>
            </x>
        </publish-options>
    </pubsub>
</iq>
```

There seem to be two distinct issues:

1. There's a duplicate bookmark for the same room (by JID). I can't think of a reason why two bookmarks for the same JID would be desirable (support for joining a room using different nicknames seems far-fetched)
2. The bookmark set by Conversations was retained, but got augmented with a `nick` element. As there was no nickname in that bookmark, it's set to `undefined`. I don't think that there should be a `nick` element at all. I believe the `nick` element is optional in all XEPs that relate to bookmarks.

**Comment by licaon-kter** [comment]
Fixed in https://github.com/conversejs/converse.js/issues/2039 ?

**Comment by jcbrand** [comment]
I know a version 7 release is overdue, and that's why bug reports for 6.0.1 still come in.

However, the current master branch has progressed so far from version 6.0.1, that bug reports for that version become less and less relevant. Given that I'm struggling to find the time to get a version 7 release out, I'm not inclined to spend even more time working on a version 6 branch and making further releases there.

Like @licaon-kter mentioned, I think this is fixed in master in #2039, so closing this as a duplicate.
